### PR TITLE
Add GetDatasetFilesTotalDownloadSize use case

### LINKED ISSUE #86: Add use case to get the total space consumed by the files of a particular dataset version
**Overview of the Feature Request**

We need a use case to get the total space consumed by the files of a particular dataset version.

This use case will allow the SPA to know the total space consumed by the files when the user selects "select all" in the files tab, without needing to retrieve all the files (without pagination, could be an expensive operation) and apply client side logic to calculate the size.

The use case may be useful for other scenarios where the total file size may be useful, for example, metrics-related applications.

We need to explore the Dataverse API to determine if an endpoint exists to support the new use case or if it needs to be created (Create API spike issue in such case).

**What kind of user is the feature intended for?**

Package consumer

**What inspired the request?**

- Discussion on https://github.com/IQSS/dataverse-frontend/pull/156

**What existing behavior do you want changed?**

None

**Any brand new behavior do you want to add to Dataverse?**

New use case for getting the total space consumed by the files of a particular dataset version

**Any open or closed issues related to this feature request?**

- https://github.com/IQSS/dataverse-frontend/pull/156

### PR DESCRIPTION
**Pull request by GPortas**
## What this PR does / why we need it:

Adds new GetDatasetFilesTotalDownloadSize use case to support SPA functionality: https://github.com/IQSS/dataverse-frontend/issues/170

## Which issue(s) this PR closes:

- Closes #86 

## Related Dataverse PRs:

- Depends on https://github.com/IQSS/dataverse/pull/9960

## Suggestions on how to test this:

- Visual inspection of the code
- Check unit and integration tests execution in GitHub actions

## Is there a release notes update needed for this change?:

Add GetDatasetFilesTotalDownloadSize use case

## Additional documentation:

None

### CODE REVIEW

File: src/files/domain/models/FileDownloadSizeMode.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum FileDownloadSizeMode {
+  ALL = 'All',
+  ORIGINAL = 'Original',
+  ARCHIVAL = 'Archival',
```
Comment (MellyGray) [inline src/files/domain/models/FileDownloadSizeMode.ts:4]: Just a quick question, so archival is with the tabular files processed, original is with the tabular files not processed and ALL I assume it means both ORIGINAL and ARCHIVAL tabular files, is it?

**Review by MellyGray — DISMISSED** [review]
LGTM!

File: src/files/domain/models/FileDownloadSizeMode.ts (hunk 1 above)
Comment (GPortas) [reply src/files/domain/models/FileDownloadSizeMode.ts:4]: That's it, as you describe!

**Comment by kcondon** [comment]
@GPortas Merge conflict, otherwise github actions test seems to pass.

**Pull request merged**